backend/remote-state/artifactory/backend.go

```diff
@@ -17,29 +18,74 @@ func New() backend.Backend {
 				Type:        schema.TypeString,
 				Required:    true,
 				DefaultFunc: schema.EnvDefaultFunc("ARTIFACTORY_USERNAME", nil),
-				Description: "Username",
+				Description: "Username for state file access",
 			},
 			"password": &schema.Schema{
 				Type:        schema.TypeString,
 				Required:    true,
 				DefaultFunc: schema.EnvDefaultFunc("ARTIFACTORY_PASSWORD", nil),
-				Description: "Password",
+				Description: "Password for state file access",
 			},
 			"url": &schema.Schema{
 				Type:        schema.TypeString,
 				Required:    true,
 				DefaultFunc: schema.EnvDefaultFunc("ARTIFACTORY_URL", nil),
-				Description: "Artfactory base URL",
+				Description: "Artfactory base URL for state file access",
 			},
 			"repo": &schema.Schema{
 				Type:        schema.TypeString,
 				Required:    true,
-				Description: "The repository name",
+				Description: "The repository name for state file access",
 			},
 			"subpath": &schema.Schema{
 				Type:        schema.TypeString,
 				Required:    true,
-				Description: "Path within the repository",
+				Description: "Path within the repository for state file access",
+			},
+			"lock_username": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("LOCK_ARTIFACTORY_USERNAME", nil),
+				Description: "username for lock file creation",
+			},
+			"lock_password": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("LOCK_ARTIFACTORY_PASSWORD", nil),
+				Description: "password for lock file creation",
+			},
+			"unlock_username": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("UNLOCK_ARTIFACTORY_USERNAME", nil),
+				Description: "username for lock file removal",
+			},
+			"unlock_password": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("UNLOCK_ARTIFACTORY_PASSWORD", nil),
+				Description: "password for lock file removal",
+			},
+			"lock_url": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("LOCK_ARTIFACTORY_URL", nil),
+				Description: "artfactory base URL for lock file access",
+			},
+			"lock_repo": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The repository for lock file access",
+			},
+			"lock_subpath": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "path within the repository for lock file access",
+			},
+			"lock_readback_wait": &schema.Schema{
```

Comment:
We prefer not to present implementation details to the use as configuration options.